airbyte-webapp/src/hooks/services/useBuildUpdateCheck.ts

```diff
@@ -0,0 +1,51 @@
+import { useEffect } from "react";
+import { useIntl } from "react-intl";
+import { fromEvent, interval, merge, throttleTime } from "rxjs";
+
+import { useNotificationService } from "./Notification";
+
+interface BuildInfo {
+  build: string;
+}
+
+const INTERVAL = 60 * 1000;
+
+/**
+ * This hook sets up a check to /buildInfo.json, which is generated by the build system on every build
+ * with a new hash. If it ever detects a new hash in it, we know that the Airbyte instance got updated
+ * and show a notification to the user to reload the page.
+ */
+export const useBuildUpdateCheck = () => {
+  const { formatMessage } = useIntl();
+  const { registerNotification } = useNotificationService();
+  useEffect(() => {
+    // Trigger the check every ${INTERVAL} milliseconds or whenever the window regains focus again
+    const subscription = merge(interval(INTERVAL), fromEvent(window, "focus"))
+      // Throttle it to maximum once every 10 seconds
+      .pipe(throttleTime(10 * 1000))
+      .subscribe(async () => {
+        try {
+          // Fetch the buildInfo.json file without using any browser cache
+          const buildInfoResp = await fetch("/buildInfo.json", { cache: "no-store" });
```

Comment:
Just double checking - our nginx config doesn't have any server-side caching set up, right? This will be a route we'll want to avoid caching on the CDN if possible.

Reply:
It has by default. In general I wanted to make this independant from whatever the infrastructure might have (miss)configured. So I thought forcing this here, makes more sense?